elpy/auto_pep8.py

```diff
@@ -11,11 +12,15 @@
     autopep8 = None
 
 
-def fix_code(code):
+def fix_code(code, directory):
     """Formats Python code to conform to the PEP 8 style guide.
 
     """
     if not autopep8:
         raise Fault('autopep8 not installed, cannot fix code.',
                     code=400)
-    return autopep8.fix_code(code, apply_config=True)
+    old_dir = os.getcwd()
+    os.chdir(directory)
+    ret = autopep8.fix_code(code, apply_config=True)
+    os.chdir(old_dir)
```

Comment:
Use `try`/`finally` here to ensure the directory is changed back even in case of an error.

Reply:
Done